### DISCUSSION TITLE
EMR - Not able to Terminate EMR cluster as Job flows are termination protected

### DISCUSSION THREAD

**Discussion by KapilTeotia1** [cloud-custodian/cloud-custodian#7397] [General]
Hi Team,

We have written an custodian policy to terminate a EMR cluster if it is running more than 12 hours. The policy is running fine and terminating the cluster perfectly fine. However if any cluster is having termination protection ON then the policy getting failed to terminate the cluster (Error Trace are below). I have gone through the available actions for EMR however could not find any which can remote the termination protection. 

Could you please advise if we can first remove the termination protection first and then terminate the cluster through custodian policy?

Error Trace:

Copy
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the TerminateJobFlows operation: Could not shut down one or more job flows since they are termination protected.

![image](https://user-images.githubusercontent.com/49716866/105682785-e02f9280-5f18-11eb-9ec4-1024673fa252.png)

**Comment by KVInventoR** [comment]
This policy works for me well
But before the best mark EMR to terminate in 1hour

`
  - name: emr_terminate_marked_clusters
    resource: emr
    description: |
      EMR - Terminate any marked clusters
    mode:
      type: periodic
      schedule: "rate(30 minutes)"
      role: *lambda_role
      tags: &tags_emr_terminate_marked_clusters
        << : *tags_basic
        Name: "emr_terminate_marked_clusters"
    filters:
      - type: marked-for-op
        op: terminate
    actions:
      - type: tag
        key: c7n_action
        value: "Terminated"
      - type: terminate
        force: true

`
